server/routes/api.js

```diff
@@ -0,0 +1,67 @@
+const express = require('express');
+const router = express.Router();
+const moment = require('moment');
+const Expense = require('../models/Expense');
+
+router.use(express.json())
+router.use(express.urlencoded({ extended: false }))
+
+// const expense = require('./expenses.json');
+//     expense.forEach(e => {
+//         let exp = new Expense ({
+//             name: e.item,
+//             amount: e.amount,
+//             date: e.date,
+//             group: e.group
+//         });
+
+//         exp.save();
+//     })
+
+
+router.get('/expenses', function (req, res) {
+    let d1;
+    req.query.d1 ? d1 = moment(req.query.d1).format('LLLL') : d1 = 0;
+    const d2 = moment(req.query.d2).format('LLLL'); //if d2 is not declared, it will automaticly become moment().
+    Expense.find({date: {"$lt": d2, "$gt": d1}}).sort({date: -1}).exec(function(err, exp) {
+        res.send(exp);
+    })
+})
+
+router.post('/expense', function(req, res) {
+    console.log('updating data');
+    const newExp = req.body;
+    const exp = new Expense({
+        name: newExp.name,
+        amount: newExp.amount,
+        group: newExp.group,
+    })
+    newExp.date ? exp.date = moment(newExp.date).format('LLLL') : exp.date = moment().format('LLLL');
```

Comment:
better practice is to write this like the following:
`exp.date =   moment(newExp.date ? newExp.date : undefined).format('LLLL')`